### ISSUE TITLE
Display full bibliography listing

### ISSUE THREAD

**Issue by Conal-Tuohy** [IUBLibTech/newton_chymistry#22]
Display a fielded view of the citation

- We need to determine display for example author name could be:
  - For example: Author: Ebehard Zetzner v. using forename/surname
- Replace element names with something more descriptive
- Provide link to fa
- (we need to find a citation with all the fields and annotate how we want the metadata to
display)
- Need to figure out the notes fields and what we want to show/not show

**Comment by ayarnell** [comment]
@Conal-Tuohy I'd like to help with this issue if I can. We talked about some of this at the meeting today. It sounds like we will want most notes to be for internal use only, but we could indicate notes for public consumption with an attribute like type="external"

**Comment by Conal-Tuohy** [comment]
Thanks @ayarnell! I am going to go ahead with the "plumbing" to load the bibliography and link its entries with the corresponding entries in the manuscripts, but I'd appreciate any help with the user interface layer. Also, do you know of any particular entries which would make good examples to work with?

**Comment by ayarnell** [comment]
I think Boyle1680 and Zetzner1659a would be good entries.

**Comment by Conal-Tuohy** [comment]
How should the "full bibliography listing" be displayed? Should it be treated like the Newton MSS themselves? Or it should it be a page all of its own? Whereabouts in the menus should it appear?

**Comment by Conal-Tuohy** [comment]
I've made the following additions and alterations:

- Added a new pipeline called [`chymistry:download-bibliography`](https://github.com/IUBLibTech/newton_chymistry/commit/68a06093322d297e18c65f199747df9e5fbc7e02#diff-5d0db68a0f2c64b48fd3ee24f1142005), to download the bibliography file from Xubmit. The Administration page has a button which invokes this pipeline by doing a HTTP POST to the URL `/download-bibliography`.
- Added a pipeline called [`chymistry:bibliography-as-html`](https://github.com/IUBLibTech/newton_chymistry/commit/68a06093322d297e18c65f199747df9e5fbc7e02#diff-a1aa6bfe04828c5927108d4fc4f79ce5R247-R258) to display the bibliography as a single HTML page. The pipeline uses a [very simple stylesheet to convert the listBibl into a web page](https://github.com/IUBLibTech/newton_chymistry/commit/68a06093322d297e18c65f199747df9e5fbc7e02#diff-9a86cc36da62c670e7cb8fff29000cfb).
- [Edited the `menus.json` file](https://github.com/IUBLibTech/newton_chymistry/commit/043775d6bcf66dac0039cb35a8541151fb6e666a#diff-efe778ea3ea70d7e050b5540ec447d5b) to add a reference to the bibliography page to the "Online Tools" menu 
 - Modified the existing pipeline for converting an individual MS into HTML, so that it first includes the entire `<listBibl>` from the bibliography into each MS file's `<sourceDesc>`, so that the full details of each citation are available for being rendered into the HTML [Each `<bibl>` in the transcript is then rendered as a hyperlink](https://github.com/IUBLibTech/newton_chymistry/commit/68a06093322d297e18c65f199747df9e5fbc7e02#diff-180d1043fe72b08714dcce6a878b3923), with a popup `title` attribute which is just the plain text of the full citation `<biblStruct>`.

So the plumbing is in place, but there are still questions of how to display an individual citation which are waiting for your input, @ayarnell and @mdalmau

**Comment by mdalmau** [comment]
Here's information, including XPaths, for the full bibliography: https://iu.app.box.com/notes/545445019301. We are working through some of the details with the team, but all agree that it's a good start.

**Comment by Conal-Tuohy** [comment]
- Use the MLA citation format


# Pop-Up Citations
This would be a javascript pop-up that remains open until closed by user (box with "x") or link selected to either view the full citation or the full text, when full text is available.  


## Book with an Author
### XPath + Citation Formatting 
/TEI/text/body/listBibl/biblStruct/monogr/author/surname, /TEI/text/body/listBibl/biblStruct/monogr/author/forename. /TEI/text/body/listBib/biblStruct/monogr/title/@type=‘short’. (Display as italicized)
/TEI/text/body/listBibl/biblStruct/monogr/imprint/publisher, /TEI/text/body/listBibl/biblStruct/monogr/imprint/date.


[View Full Citation]    [View Full Text: The Value of Ref: /TEI/text/body/listBibl/biblStruct/monogr/title/@ref]


If /TEI/text/body/listBibl/biblStruct/monogr/title/@ref is NULL, then don't print [View Full Text].


### Rendered View
Boyle, Robert. The origine of formes and qualities. H. Hall for R. Davis, 1666.
[View Full Citation]    [View Full Text]


## Book without an Author
To keep things simple, I would render the citation for anthologies as a book without an author.  It may be cumbersome for Con to figure out whether there's an editor/contributor/printer or however else we attribute an anthologized work.  


### XPath + Citation Formatting 
/TEI/text/body/listBib/biblStruct/monogr/title/@type=‘short’. (Display as italicized)
/TEI/text/body/listBibl/biblStruct/monogr/imprint/publisher, /TEI/text/body/listBibl/biblStruct/monogr/imprint/date.


[View Full Citation]    [View Full Text: The Value of Ref: /TEI/text/body/listBibl/biblStruct/monogr/title/@ref]


If /TEI/text/body/listBibl/biblStruct/monogr/title/@ref is NULL, then don't print [View Full Text].


### Rendered View
Theatrum Chemicum Britannicum. Nath. Brooke, 1652.
[View Full Citation]    [View Full Text]


# List of All Citations (Full Bibliography)
The list of all citations should be organized A-Z by biblStruct@xml:id. The list should be fielded.  When a value is not present, do not list the field.  Invite Meridith and Bill to author an introduction to the Bibliography (or if Bill only, credit Meridith, Meagan and Amy).  


## Fielded List: XPath + Citation Formatting 


### Author: 
*Add a slash (/) between every name / forename surname.  Some contain 3 or more authors.
/TEI/text/body/listBibl/biblStruct/monogr/author/name / 
/TEI/text/body/listBibl/biblStruct/monogr/author/forename 
/TEI/text/body/listBibl/biblStruct/monogr/author/surname
### Title (Brief):
*Display as italicized 
/TEI/text/body/listBib/biblStruct/monogr/title/@type=‘short’ 
### Title Page:
/TEI/text/body/listBibl/biblStruct/monogr/title/@level='m'
### Contributors:
(check for respStatmt: /TEI/text/body/listBibl/biblStruct/monogr/imprint/respStmt
/TEI/text/body/listBibl/biblStruct/monogr/imprint/respStmt/orgName (Print value of /TEI/text/body/listBibl/biblStruct/monogr/imprint/respStmt/resp)
### Publisher:
/TEI/text/body/listBibl/biblStruct/monogr/imprint/publisher
### Publication Place:
/TEI/text/body/listBibl/biblStruct/monogr/imprint/pubPlace
### Publication Year:
/TEI/text/body/listBibl/biblStruct/monogr/imprint/date
### Link to Fulltext:
Print value of ref: /TEI/text/body/listBibl/biblStruct/monogr/title/@ref 


## Fielded List: Rendered View


### Author:
EIRENAEUS PHILALETHES / William COOPER
### Editor:
TBD
### Title (Brief):
Ripley Reviv'd (Display as italicized)
### Title Page:
*<hi> tags used to indicate italics within the title tag
DOMVS NATVRÆ│ Ripley Reviud│Nil decit nisi Clavis │ Printed for William Cooper at the
              Pelican in little Britain 1678 [UNCORRECTED] Ripley Reviv'd: or, an exposition upon
              Sir George Ripley's Hermetico-Poetical Works, containing .. discoveries of the most
              hidden secrets of the ancient philosophers .. Written by Eirenæus Philalethes, etc.
              (Experiments for the preparation of the sophick Mercury .. for the Philosopher's Stone
              .. by Eirenæus Philalethes.) [Edited by W. Cooper].
### Contributors:
Thomas Ratcliff and Nathan Thompson (Printer)
### Publication Place:
London
### Publication Year:
1677
### Link to Fulltext:
https://quod.lib.umich.edu/e/eebo/A61326.0001.001?rgn=main;view=toc


### TEI Bibliography Corrections by Newton Encoders
xml:id values for the title need to be normalized: Mix-case (some are all uppercase, others start with a lowercase letter which will mess up sorting)
need to separate editors from authors 
title@ref should be placed in the short title, not the QFT title (?)
QFT need to be corrected
Names need to be normalized (some are all in uppercase)
In respStatmt, if there's no organization, remove respStmt
Some citations are commented out ... why? 
Place names will be normalized to modern day places

**Comment by Conal-Tuohy** [comment]
What should be displayed for `<author>` elements which do not contain `<name>`, `<forename>`, or `<surname>` elements, but instead contain text nodes as direct children (e.g. `<author>Various</author>` or `<author>Royal Society</author>` or `<author>Anonymous</author>`? I've taken the instructions above to imply that these `<author>` elements should be omitted from the display, but I'd like a confirmation that that's correct. @mdalmau ?

**Comment by Conal-Tuohy** [comment]
In the citation popups, I decided to render author names even where they aren't divided into a surname and forename. The instructions above don't cover that case. Let me know if that's not the desired behaviour, please, @mdalmau

**Comment by Conal-Tuohy** [comment]
Ready for this to be reviewed

**Comment by mdalmau** [comment]
I have emailed the project team for their input on 8 Dec 2019: 

Hi, all, (@jawalsh, @wehooper) 

Many weeks ago, Con implemented the listing of the full bibliography on the new Newton site:
https://newton.dlib.indiana.edu/bibliography

Please review to make sure this is to your satisfaction.  

The main things I notice as problematic are fixes we need to make on our end but should render correctly once the bibliography XML file is re-indexed:

* Unlisted names should be removed from the encoding.  If we don’t know the Publisher or Contributor, we  should not encode that with a placeholder value like “unlisted” or with the role like “printer.”  

* The links to the full texts are a bit of a mess.  Some are broken.  Others are too long.  We should be using the parma-links provided by the digital sources like Internet Archive and not the URL that displays on the browser.  This can be fixed easy-enough, though.  

Bill, we had discussed some other changes, but I was waiting to note those as part of your full site review.  If you have other questions or editions, please let us know.  We need to make sure Con is handling all cases for display, even if the XML element isn’t populated.   

-------

@Conal-Tuohy, I am sure that you are documenting this, but we would need to know how to refresh the indexes so that when we make these changes, the application picks  up the new  encoding.  This is true for  the bibliography  and the mss.  

Thanks!
—Michelle

**Comment by mdalmau** [comment]
Amy Yarnell responded to the email above: In the most recent version of the bibliography on Xubmit from 12/8 all the “unlisted” should be gone. I checked that before I left. Most of the links should be fixed too, I was correcting those as I went. There might still be some wonky ones in the last batch that need title page transcriptions corrected. 

@wehooper and team, reviewed the bibliography and responded: Hi Con and Michelle,
 
AT the meeting, we’ve just looked through the entries in the new authoritative bibliography, and we think it looks fine.
 
We can take over the problem of implementing the blackletter font and red highlights if not completed and any other developments of this component.
 
Wally and the team
--------------------
I can create a separate issue for the font rendering and assign it to someone on the Project team.  We can close this issue!
